tests/x11/window_system.pm

```diff
@@ -4,21 +4,31 @@
 #
 # Copying and distribution of this file, with or without modification,
 # are permitted in any medium without royalty provided the copyright
-# notice and this notice are preserved.  This file is offered as-is,
+# notice and this notice are preserved. This file is offered as-is,
 # without any warranty.
 
-# Summary: Information about current window system
+# Summary: Show information about current session (window system)
 # Maintainer: Jan Baier <jbaier@suse.cz>
 
-use base "x11test";
+use base "opensusebasetest";
 use strict;
+use warnings;
 use testapi;
 
 sub run {
-    x11_start_program('xterm');
-    my $window_system = script_output('echo $XDG_SESSION_TYPE');
-    script_run('exit', 0);
-    record_info("$window_system", "Current window system is $window_system");
+    my $self = shift;
+    $self->select_serial_terminal;
+
+    my $session_id   = script_output "loginctl list-sessions | awk '/$testapi::username/ {print \$1};'";
```

Comment:
each `script_output` call is pretty complex thing with waiting for output several times. Because you need `session_id`  only for bypass it to next `script_output` can you try to combine them into single `script_output` call ?

Reply:
Sure.